portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,50 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
+
 import java.io.IOException;
+import java.util.List;
+import java.util.ArrayList;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/data-comments")
 public class DataServlet extends HttpServlet {
-
+  private List<String> comments;
+  private Gson gson;
+  @Override
+  public void init(){
+    gson = new Gson();
+    comments = new ArrayList<>();
+  }
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    //response.setContentType("text/html;");
+    //response.getWriter().println("Hello!!");
+    String jsonComments = gson.toJson(comments);
+    response.setContentType("application/json;");
+    response.getWriter().println(jsonComments);
+  }
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = getParameter(request, "commentText", "error");
```

Comment:
instead of adding the string "error" on failure, maybe it would be better to just fail entirely. maybe something like throw IllegalArgumentException instead?

Reply:
Sounds good, will change that!